Services/Administration/OJS.Services.Administration.Business/Implementations/ProblemsBusinessService.cs

```diff
@@ -176,6 +176,21 @@ public async Task<bool> UserHasProblemPermissions(int problemId, string? userId,
             return await this.contestsBusiness.UserHasContestPermissions(problem.ContestId, userId, isUserAdmin);
         }
 
+        public async Task ReevaluateProblemsByOrderBy(int contestId)
+        {
+            var orderByIndex = 0;
+
+            await this.contestsData.GetProblemsById(contestId)
+                                    .OrderBy(p => p.OrderBy)
+                                    .ForEachAsync(p =>
+                                    {
+                                        p.OrderBy = ++orderByIndex;
+                                        this.problemsData.Update(p);
+                                    });
+
+            await this.problemsData.SaveChanges();
```

Comment:
Use `SaveChangesAsync()`

Reply:
Each dataService inherits DatService which has the following implementation:
 public virtual async Task SaveChanges()
```suggestion
            await this.problemsData.SaveChanges();
```
            => await this.db.SaveChangesAsync();